### ISSUE TITLE
How do I run more iteration with LightGBMTuner with a study?

### ISSUE THREAD

**Issue by wil70** [optuna/optuna#5272]
### Don't use GitHub Issues to ask support questions.

Hello, 

**1) How do I run more iteration with LightGBMTuner with a study?**
I'm attempting to perform an additional 900 iterations after the initial 100, but I'm encountering difficulties. I adapted an example from Optuna + LightGBM, replacing the training with LightGBMTuner. Here is the code:

```
import optuna
import optuna.integration.lightgbm as lgb
import numpy as np
from lightgbm import early_stopping
from lightgbm import log_evaluation
import sklearn.datasets
from sklearn.model_selection import KFold
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

if __name__ == "__main__":
    data, target = sklearn.datasets.load_breast_cancer(return_X_y=True)
    #dtrain = lgb.Dataset(data, label=target)
    #train_x, valid_x, train_y, valid_y = train_test_split(data, target, test_size=0.25)
    train_x, test_x, train_y, test_y = train_test_split(data, target, test_size=0.10)
    train_x, valid_x, train_y, valid_y = train_test_split(train_x, train_y, test_size=0.25)
    dtrain = lgb.Dataset(train_x, label=train_y)
    dval = lgb.Dataset(valid_x, label=valid_y)

    params = {
        "objective": "binary",
        "metric": "binary_error",
        "verbosity": -1,
        "boosting_type": "dart",
    }

    modelDir="QuickTestTunerModel"

    try:
        study = optuna.load_study(study_name='study-wil10_8_v2', storage='sqlite:///'+modelDir+'\\wil10_8_v2.db')
    except KeyError:
        study = optuna.create_study(study_name='study-wil10_8_v2', storage='sqlite:///'+modelDir+'\\wil10_8_v2.db', direction="minimize")

    tuner = lgb.LightGBMTuner(
            params,
            dtrain,
            valid_sets=[dval], #####
            study=study,
            num_boost_round=100,
            #folds=KFold(n_splits=3),
            callbacks=[early_stopping(100), log_evaluation(100)],
            model_dir=modelDir,
            keep_training_booster =True,
            #return_cvbooster=True
    )
    

    tuner.run()
    tuner.run()
    
    # Run an additional 100 trials
    #study.optimize(n_trials=100)
    #tuner.run()

    print("Best score:", tuner.best_score)
    print("Best booster score:", tuner._get_booster_best_score)
    best_params = tuner.best_params
    print("Best params:", best_params)
    print("  Params: ")
    for key, value in best_params.items():
        print("    {}: {}".format(key, value))

 
    # Save the best model
    tuner.get_best_booster().save_model("QuickTestTunerModel\\bestModel.txt")

    # Load the saved model
    bst = tuner.get_best_booster() #lgb.Booster(model_file='bestModel.txt')

    preds = bst.predict(train_x)
    preds = np.rint(preds)
    accuracy = accuracy_score(train_y, preds)
    print("train accuracy: "+str(accuracy))
    
    preds = bst.predict(valid_x)
    preds = np.rint(preds)
    accuracy = accuracy_score(valid_y, preds)
    print("valid accuracy: "+str(accuracy))

    # Predict on the testing data
    preds = bst.predict(test_x)
    preds = np.rint(preds)
    accuracy = accuracy_score(test_y, preds)
    print("test accuracy: "+str(accuracy))
```

The output look like this

```
Best score: 0.0234375
Best booster score: <bound method _BaseTuner._get_booster_best_score of <optuna.integration._lightgbm_tuner.optimize.LightGBMTuner object at 0x000001FAAAEA5460>>
Best params: {'objective': 'binary', 'metric': 'binary_error', 'verbosity': -1, 'boosting_type': 'dart', 'feature_pre_filter': False, 'lambda_l1': 0.0, 'lambda_l2': 0.0, 'num_leaves': 31, 'feature_fraction': 0.8, 'bagging_fraction': 1.0, 'bagging_freq': 0, 'min_child_samples': 20}
  Params:
    objective: binary
    metric: binary_error
    verbosity: -1
    boosting_type: dart
    feature_pre_filter: False
    lambda_l1: 0.0
    lambda_l2: 0.0
    num_leaves: 31
    feature_fraction: 0.8
    bagging_fraction: 1.0
    bagging_freq: 0
    min_child_samples: 20
train accuracy: 1.0
valid accuracy: 0.9765625
test accuracy: 0.9824561403508771
```
The second time I run the program it load from the sqllite database but it doesn't run more iterations. I tried adding an extra tuner.run() and other things but I couldn't make it work.

Output of the second run:
```
 0%|                                                                                                                                                                                                                                                                                                                                                                                                                        | 0/7 [00:00<?, ?it/s]
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                       | 0/20 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                       | 0/10 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                        | 0/6 [00:00<?, ?it/s]
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                       | 0/20 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                        | 0/5 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                        | 0/7 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                       | 0/20 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                       | 0/10 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                        | 0/6 [00:00<?, ?it/s]
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                       | 0/20 [00:00<?, ?it/s] 
  0%|                                                                                                                                                                                                                                                                                                                                                                                                                        | 0/5 [00:00<?, ?it/s] 
Best score: 0.0234375
Best booster score: <bound method _BaseTuner._get_booster_best_score of <optuna.integration._lightgbm_tuner.optimize.LightGBMTuner object at 0x000002329EA68410>>
Best params: {'objective': 'binary', 'metric': 'binary_error', 'verbosity': -1, 'boosting_type': 'dart', 'feature_pre_filter': False, 'lambda_l1': 0.0, 'lambda_l2': 0.0, 'num_leaves': 31, 'feature_fraction': 0.8, 'bagging_fraction': 1.0, 'bagging_freq': 0, 'min_child_samples': 20}
  Params:
    objective: binary
    metric: binary_error
    verbosity: -1
    boosting_type: dart
    feature_pre_filter: False
    lambda_l1: 0.0
    lambda_l2: 0.0
    num_leaves: 31
    feature_fraction: 0.8
    bagging_fraction: 1.0
    bagging_freq: 0
    min_child_samples: 20
train accuracy: 0.9895833333333334
valid accuracy: 1.0
test accuracy: 1.0

```

Despite these attempts, the program doesn't run more iterations. How can I achieve this?

**2) Potential Issues:**
Upon the second run, I observed differences in training accuracy. Initially at 1, it decreased later. Meanwhile, validation and test accuracy increased to 1. This behavior seems unusual. Any insights into this?

Thank you for your assistance.

--Wil
